src/rebar_prv_plugins.erl

```diff
@@ -0,0 +1,125 @@
+-module(rebar_prv_plugins).
+
+-behaviour(provider).
+
+-export([init/1,
+         do/1,
+         format_error/1]).
+
+-include("rebar.hrl").
+-include_lib("providers/include/providers.hrl").
+
+-define(PROVIDER, plugins).
+-define(DEPS, []).
+
+-spec init(rebar_state:t()) -> {ok, rebar_state:t()}.
+init(State) ->
+    State1 = rebar_state:add_provider(
+            State,
+            providers:create([
+                    {name, ?PROVIDER},
+                    {module, ?MODULE},
+                    {bare, true},
+                    {deps, ?DEPS},
+                    {example, "rebar3 plugins"},
+                    {short_desc, "List or uprade plugins"},
+                    {desc, "List or upgrade plugins"},
+                    {opts, []}])),
+    {ok, State1}.
+
+-spec do(rebar_state:t()) -> {ok, rebar_state:t()} | {error, string()}.
+do(State) ->
+    case rebar_state:command_args(State) of
+        ["list"] ->
+            list(State);
+        ["upgrade", Plugin] ->
+            upgrade(Plugin, State);
+        [] ->
+            list(State)
```

Comment:
to match `upgrade` for deps/packages and also `unlock`, we might want to support `upgrade` without any argument that means "upgrade every plugin"